cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -143,6 +143,18 @@ type ClusterStateRegistry struct {
 	scaleUpFailures map[string][]ScaleUpFailure
 }
 
+// NodeGroupHealth contains information about the health of the node group.
+type NodeGroupHealth struct {
```

Comment:
Hmm, I still don't understand why we need these two types at all. Either I'm really not seeing something here, or the `ErrorInfo` parameter seems completely unused other than passing it through from `GetNodeGroupHealth` to `IsNodeGroupSafeToScaleUp` (after which it's not used anymore).

Wouldn't the PR work exactly the same if we didn't change `IsNodeGroupSafeToScaleUp` and `IsNodeGroupHealthy` at all, just with less complexity? From a high level, the "safety" determination is based on backoff info, rather than being an input to backoff - so we shouldn't need to keep this error around to feed it to the new backoff interface, right?

Reply:
Sorry about that. It was needed but the needed part didn't get in the this commit after splitting.
I've added a line where it is used.